pkg/workceptor/kubernetes.go

```diff
@@ -606,7 +654,17 @@ func (kw *kubeUnit) SetFromParams(params map[string]string) error {
 			}
 		}
 	}
-	ked.Params = combineParams(kw.baseParams, userParams)
+	if userPodSpec != "" && (userParams != "" || userCommand != "" || userImage != "") {
+		return fmt.Errorf("params kube_command, kube_image, kube_params not compatible with secret_kube_podspec")
+	}
+	ked.Command = userCommand
```

Comment:
For clarity of which things are happening in which circumstance, what do you think of arranging this more like:

```
	if userPodSpec == "" {
		ked.Command = userCommand
		ked.Image = userImage
		ked.Params = combineParams(kw.baseParams, userParams)
	} else {
		ked.KubePodSpec = userPodSpec
		kw.baseParams = ""
	}
```

Reply:
yep that is clearer, fixed!